### ISSUE TITLE
Automatically delete a note when it's content is empty

### ISSUE THREAD

**Issue by michaelkolesidis** [Automattic/simplenote-macos#1064]
### What
When I delete the whole text content of a note, the note stays in the database as an empty note and in order to be removed, one has to manually trash it. I propose that whenever a user deletes all the text content of a note and then focuses on another element (click on another note, perfoms another action elsewhere), the note is deleted automatically. It is a common feature of note taking apps, implemented in Apple Notes for Mac for example.

### Why
It will save a lot of time, and provide an easy way to delete unwanted notes.

**Comment by charliescheer** [comment]
Thank you for the suggestion.  We discussed this change internally and decided that we are not going to make notes automatically delete if they aren't new and their content is removed.  Primarily we decided not to do this because it makes accessing the version history of a note more difficult if the note is automatically moved to the trash empty as all notes deleted this way will look the same, it makes recovery very difficult.
